statistics/histogram.go

```diff
@@ -2280,3 +2274,44 @@ func MergePartitionHist2GlobalHist(sc *stmtctx.StatementContext, hists []*Histog
 	}
 	return globalHist, nil
 }
+
+// ColLoadedStatus indicates the status of column statistics
+type ColLoadedStatus struct {
+	wasLoaded         bool
+	cmsEvicting       bool
+	topnEvicting      bool
+	histogramEvicting bool
+}
+
+// NewColFullLoadStatus returns the status that the column fully loaded
+func NewColFullLoadStatus() ColLoadedStatus {
+	return ColLoadedStatus{
+		wasLoaded: true,
+	}
+}
+
+// WasLoaded indicates whether the column's statistics was loaded from storage before.
+// Note that `wasLoaded` only can be set in initializing
+func (s ColLoadedStatus) WasLoaded() bool {
```

Comment:
IsLoaded is better, we are checking the current status instead of the historical one.

Reply:
updated.